cmd/machine-controller-manager/app/options/options.go

```diff
@@ -51,17 +51,18 @@ func NewMCMServer() *MCMServer {
 		// Part of these default values also present in 'cmd/cloud-controller-manager/app/options/options.go'.
 		// Please keep them in sync when doing update.
 		MachineControllerManagerConfiguration: machineconfig.MachineControllerManagerConfiguration{
-			Port:                    10258,
-			Namespace:               "default",
-			Address:                 "0.0.0.0",
-			ConcurrentNodeSyncs:     5,
-			ContentType:             "application/vnd.kubernetes.protobuf",
-			NodeConditions:          "KernelDeadlock,ReadonlyFilesystem,DiskPressure",
-			MinResyncPeriod:         metav1.Duration{Duration: 12 * time.Hour},
-			KubeAPIQPS:              20.0,
-			KubeAPIBurst:            30,
-			LeaderElection:          leaderelectionconfig.DefaultLeaderElectionConfiguration(),
-			ControllerStartInterval: metav1.Duration{Duration: 0 * time.Second},
+			Port:                      10258,
+			Namespace:                 "default",
+			Address:                   "0.0.0.0",
+			ConcurrentNodeSyncs:       5,
+			ContentType:               "application/vnd.kubernetes.protobuf",
+			NodeConditions:            "KernelDeadlock,ReadonlyFilesystem,DiskPressure",
+			MinResyncPeriod:           metav1.Duration{Duration: 12 * time.Hour},
+			KubeAPIQPS:                20.0,
+			KubeAPIBurst:              30,
+			LeaderElection:            leaderelectionconfig.DefaultLeaderElectionConfiguration(),
+			ControllerStartInterval:   metav1.Duration{Duration: 0 * time.Second},
+			ExternalMachineController: false,
```

Comment:
Yes, we could do this. However, this would partly depend on how our migration logic would look like. For example if we continue working with machine objects the way they are today and ignore the `machineClassType` field to fetch the machine classes, we might need some logic like this. 

But yes, I could revert this change for now and consider reconciling based on `machineClassType` ref in machine objects and reintroduce this change if necessary for migration. Reverting this commit.

Reply:
Actually, this has less to do with migration logic and more to do with backward compatibility. For backward compatibility, you anyway need to check the machineclass kind. Just extend it to filter out the generic machineclass.

Reply:
Yes Amshu, this check kind of already exists today - https://github.com/prashanth26/machine-controller-manager/blob/spliting-controllers2/pkg/controller/machine_util.go#L259-L260. Just reworded the log message a little. 

So the current MCM will not process the genericMachineClass ref objects with the current code as well. I forgot to mention this earlier.

Reply:
A minor improvement, currently the filtering is happening during reconciliation. It might be better to do the filtering before enqueing, no?

Reply:
Yes, I think that should be better. Shall make that change.

Reply:
I have made the changes here - https://github.com/gardener/machine-controller-manager/pull/460/commits/61da081cbade894e5cd3144bc40b7fc97adf54cc.